code-challenges.test.js

```diff
@@ -15,24 +15,97 @@
 // a) Create a test with an expect statement using the variable provided.
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
+test('removeAndShuffle - colors1', () => {
+  const shuffledColors1 = removeAndShuffle([...colors1]);
+  expect(shuffledColors1).toContain("yellow");
+  expect(shuffledColors1).toContain("blue");
+  expect(shuffledColors1).toContain("pink");
+  expect(shuffledColors1).toContain("green");
+  expect(shuffledColors1).toHaveLength(4);
+});
+
+test('removeAndShuffle - colors2', () => {
+  const shuffledColors2 = removeAndShuffle([...colors2]);
+  expect(shuffledColors2).toContain("chartreuse");
+  expect(shuffledColors2).toContain("indigo");
+  expect(shuffledColors2).toContain("periwinkle");
+  expect(shuffledColors2).toContain("ochre");
+  expect(shuffledColors2).toContain("aquamarine");
+  expect(shuffledColors2).toContain("saffron");
+  expect(shuffledColors2).toHaveLength(6);
```

Comment:
There are some things to consider on this problem.  In general, your logic in the function looks great, but the first item in the array is not being removed.  This could be causing some of your test to fail.  I am also curious about the structure of your test.  It certainly works, however it may be adding some extra complexities.  I always recommend creating a template from the Jest section in the syllabus to just plug and play the different pieces to put it together.